app_check/src/FirebaseAppCheck.cs

```diff
@@ -79,8 +87,18 @@ public sealed class FirebaseAppCheck {
   ///
   /// This method should be called before initializing the Firebase App.
   public static void SetAppCheckProviderFactory(IAppCheckProviderFactory factory) {
+    if (appCheckFactory == factory) return;
+
     appCheckFactory = factory;
-    // TODO(amaurice): Clear the provider map when the factory changes?
+    // Clear out the Providers that were previously made
```

Comment:
Just extend this comment a bit to explain why the map is being cleared.

// Clear out the Providers that were previously made. When GetToken fails to find a provider in this map it will use appCheckFactory to create a new provider.

Reply:
Done